### ISSUE TITLE
preprocessCore thread failure

### ISSUE THREAD

**Issue by lappazos** [bmbolstad/preprocessCore#19]
So just to fully understand - beside the option of disabling threads using "BiocManager::install("preprocessCore", configure.args="--disable-threading")", there is no solution? because as shown below, it produces worse performance time.

To Summarize Test results above

| Method  |  Elapsed Time  (in Seconds) |
| ---| ---|
| Using threaded openblas and preprocessCore with --disable-threading | 799.498  |
| Using normal R BLAS and preprocessCore with --disable-threading |822.104|
|Using normal R BLAS and preprocessCore with threading enabled (but no explicit R_THREADS and so only 1 thread) |  825.097   |
| Using normal R BLAS and preprocessCore with threading enabled (R_THREADS=2) |470.255 |
| Using normal R BLAS and preprocessCore with threading enabled (R_THREADS=3 on a dual core machine) |437.169 |

In other words for this particular workflow as tested on this machine, there was a small benefit to using a threaded openblas, but using a properly configured preprocessCore and it's own internal threading provides greater benefits.  

Of course these statements may not generalize. YMMV

_Originally posted by @bmbolstad in https://github.com/bmbolstad/preprocessCore/issues/1#issuecomment-326821618_

**Comment by bmbolstad** [comment]
I do not currently have a resolution to this issue.   An attempted fix had to be backed out because of issues with Apple M1 testing and I have not had time to come back and look at things recently.

**Comment by shirleyxueli41** [comment]
I tried options(mc.cores = 4) (or setting it to the number of cores available), and that solved my issue.
